File: tobaco/lib/Models/Pedido.dart
// ignore_for_file: file_names

import "package:tobaco/Models/Cliente.dart";
import "package:tobaco/Models/Producto.dart";



class Pedido {
  int? id;
  Cliente cliente;
  List<Producto> productos;
  int total;
  DateTime fecha;

  Pedido({
    this.id,
    required this.cliente,
    required this.productos,
    required this.total,
    required this.fecha,
  });
}

File: TobacoBackend/TobacoBackend/Domain/Models/Pedido.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TobacoBackend.Domain.Models
{
    public class Pedido
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        public int ClienteId { get; set; }

        [ForeignKey("ClienteId")]
        public Cliente Cliente { get; set; }

        public List<PedidoProducto> PedidoProductos { get; set; } = new List<PedidoProducto>();

        [Required]
        public decimal Total { get; set; }

        [Required]
        public DateTime Fecha { get; set; }
        public MetodoPagoEnum MetodoPago { get; set; }
    }
}
